components/validate.js

```diff
@@ -0,0 +1,77 @@
+class Validator {
+    validateSet(set) {
+        var durationCheck = false;
+        var notesCheck = false;
+        var repetitionCheck = false;
+        var weightCheck = false;
+        if (set != undefined && set != null) {
+            if (set.duration > 0) {
+                durationCheck == true;
+            }
+            if (set.notes.length < 100) {
+                notesCheck == true;
+            }
+            if (set.repetitions > 0) {
+                repetitionCheck == true;
+            }
+            if (Number.isInteger(set.weight)) {
+                weightCheck == true;
```

Comment:
1) Same as notes. This field is optional. Validator shouldn't fail if it's absent.
2) Weight shouldn't be integer only, what if I want to have `10.5` kg?

Reply:
According to the document it cannot have a decimal point. It is mentioned in BR10.